src/array.cpp

```diff
@@ -23,12 +23,12 @@
 namespace sparrow
 {
     array::array(ArrowArray&& array, ArrowSchema&& schema)
-        : p_array(array_factory(arrow_proxy(std::move(array), std::move(schema))))
+        : p_array(array_factory(arrow_proxy(std::forward<ArrowArray>(array), std::forward<ArrowSchema>(schema))))
```

Comment:
`std::move` is correct here, ArrowArray and ArrowSchema are not template parameters.